File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        freq = [0]*26
        n = len(s)
        if n==1:
            return 1
        i = 0
        j = i
        maxlen = 0; maxfreq = 0
        while (j<n):
            winsize = j-i+1
            freq[ord(s[j])-ord('A')] += 1
            maxfreq = max(maxfreq,freq[ord(s[j])-ord('A')])
            if winsize-maxfreq <= k:
                maxlen = max(maxlen,j-i+1)
                j+=1
            else:
                freq[ord(s[i])-ord('A')]-=1
                i+=1
                j+=1
        return maxlen

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
    //    vector<int>mpp(26,0);
        vector<int>mpp(26,0);
        int n = s.size();
        if (n==1)
            return 1;
        
        int i = 0;
        int j = i;
        int maxLen = 1;

        int maxFreq = 0;
        while ( j<n ) {
            int winsize = j-i+1;
            mpp[s[j]-'A']++;
            maxFreq=max(maxFreq,mpp[s[j]-'A']);
            if (winsize-maxFreq <= k) {
                // mpp[s[j]]++;
                maxLen = max(maxLen,j-i+1);
                j++;
            }
            else {
                mpp[s[i]-'A']--;
                i++;
                j++;
            }
            // j++;
        }
        return maxLen;
    }
};